README-DEV.md

```diff
@@ -144,6 +144,49 @@ Please take into account the following when sending a PR:
     - make sure there is a license header in all the files
 
 
+## How To Add a new Module
+- Use [starport](https://github.com/tendermint/starport) command to scaffold the module.
+- Follow the [README.md](scripts/starport/README.md).
+- Have a look at the scripts and commands used for generation of existing modules and do it in a similar way
+  (for example [PKI module commands](scripts/starport/upgrade-0.44/07.pki_types.sh)).
+- Adjust the generated code
+  - correct REST endpoint: `/dcl` instead of `/zigbee-alliance/distributedcomplianceledger` in `proto/<module>/query.proto`
+  - add message validation as annotations (`validate` tags) in `proto/<module>/tx.proto`
+  - add `(cosmos_proto.scalar) = "cosmos.AddressString"` annotation for all fields with address/account type (such as `signer` or `owner`).
+  - fix types if needed in `proto/<module>/<entity>.proto` files
+    - Note1: `unit64` will be returned as string if the output is a JSON format. So, it's better to use `uint64` only when it's really `uint64`. 
+    - Note2: for `uint16` type: use `int32` during starport scaffolding, and add custom validation (annotations above) to check the lower and upper bounds.
+    - Note3: for `uint32` type: use `int32` during starport scaffolding, then replace it by `uint32` in .proto files, re-generate the code and fix compilation errors.
+  - build proto (for example `starport chain build`). Fix compilation errors if any.
+  - **Note**: colons (`:`) are part of subject-id in PKI module, but colons are not allowed in gRPC REST URLs by default.
+    `allow_colon_final_segments=true` should be used as a workaround.
+    So, make sure that `runtime.AssumeColonVerbOpt(true)` in `/x/pki/types/query.pb.gw.go`. 
+    It's usually sufficient to revert the generated changes in `/x/pki/types/query.pb.gw.go`.
+- Call `validator.Validate(msg)` in `ValidateBasic` methods for all generated messages
+- Improve `NotFound` error processing:
+    - replace `status.Error(codes.InvalidArgument, "not found")` to `status.Error(codes.NotFound, "not found")` in every generated `grpc_query_xxx.go` to return 404 error in REST.
+    - Add the following to every `cli/query_xxx.go` to not throw an error and help for not found entities in CLI
+    ``` 
+     if cli.IsNotFound(err) {
+         return clientCtx.PrintString(cli.NotFoundOutput)
+     }
+     if err != nil {
+         return err
+     }
+    ```
+- Implement business logic in `msg_server_xxx.go`
+- Add unit tests (see other modules for reference)
+- Add CLI-based integration tests to `integration_tests/cli/<module>` (see other modules for reference)
+- Add gRPC/REST-based integration tests to `integration_tests/grpc_rest/<module>` (see other modules for reference)
+
+## How To Make Changes in Data Model for Existing Modules
+- Do the changes in `.proto` files. Never change `.pb` file manually.
```

Comment:
I think makes sense to emphasize that more like: **Never change `.pb` file manually.**